gui-tk/csvImport.tcl

```diff
@@ -96,7 +96,12 @@ proc CSVImportDialog {} {
             csvDialog.requestRedraw
         }
     }
-    set filename [tk_getOpenFile -filetypes {{CSV {.csv}} {All {.*}}} -initialdir $workDir]
+    
+    if [string length $url] {
+	   set filename [csvDialog.loadWebFile $url]
```

Comment:
I would have arranged loadWebFile to call loadFile, then delete the temporary file, all within C++, preferably in a destructor. I would eliminate the need to pass $url as a flag to delete the temp file within TCL.

Reply:
I couldn't accomplish this today and decided to move on to implementing the other suggestions for improving the code you made below.